### PR TITLE
Document Elasticsearch Enterprise

### LINKED ISSUE #3210: Document Elasticsearch Enterprise
### Where on docs.platform.sh should be changed?

https://docs.platform.sh/add-services/elasticsearch.html

### What exactly should be updated?

Elasticsearch Enterprise is going to be available to Enterprise clients, so it needs to be reflected in the docs.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AnouckColson**
## Why

Closes #3210

### CODE REVIEW

File: docs/src/add-services/elasticsearch.md [hunk 1]
```diff
@@ -22,11 +23,23 @@ See the [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsea
...
+but they're at their end of life and are no longer receiving security updates from upstream.
+Due to a licensing change, newer versions of Elasticsearch aren't included in any Platform.sh plan.
 
-{{% image-versions-legacy "elasticsearch" %}}
+To avoid the additional cost of subscribing to a recent version of Elasticsearch,
+you can switch to one of the latest, free versions of [OpenSearch](./opensearch.md).
+To do so, follow the same procedure as for [upgrading](#upgrading).
+
+If you want to keep using Elasticsearch,
+you need to add a recent version at an additional cost [through Sales](https://platform.sh/contact/).
```
Comment (remilejeune2) [inline docs/src/add-services/elasticsearch.md:35]: When reading this sentence, it sounds strange to me.
"If you want to keep using Elasticsearch, you need to add it to your plan at an additional cost through Sales" ?

**Review by remilejeune2 — APPROVED** [review]

File: docs/data/registry.json [hunk 2]
```diff
@@ -124,10 +124,16 @@
...
-      "supported": []
+      "supported": [
+        "8.5",
+        "7.17"
+      ]
     },
     "versions-dedicated-gen-2": {
-      "supported": [],
+      "supported": [
+        "8.15",
```
Comment (chadwcarlson) [inline docs/data/registry.json:134]:
```suggestion
        "8.5",
```

Given this comment?

**Review by chadwcarlson — APPROVED** [review]
Approved aside from the version comment

**Review by chadwcarlson — APPROVED** [review]

File: docs/src/add-services/elasticsearch.md [hunk 3]
```diff
@@ -22,28 +23,48 @@ See the [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsea
 
 ## Supported versions
 
-Due to a licensing change, Elasticsearch versions after 7.10 aren't supported.
-For newer versions, use [OpenSearch](./opensearch.md) instead.
-To switch to OpenSearch, follow the same procedure as for [upgrading](#upgrading).
+Elasticsearch is now a premium service.
+This means that from version 7.17 onwards, you need to add Elasticsearch to your project at an additional cost.
```
Comment (remilejeune2) [inline docs/src/add-services/elasticsearch.md:27]: It's >= 7.11 or > 7.10. As it, we could think that 7.11, 7.12 up to 7.16 are free of charge.

**Pull request merged**